File: src/Python/251.Flatten2DVector.py

class Vector2D:
    def __init__(self, vec2d):
        self.listOfList = vec2d
        self.listIndex = 0
        self.elemIndex = 0


    def next(self):
        val = self.listOfList[self.listIndex][self.elemIndex]
        self.elemIndex += 1
        return val


    def hasNext(self):
        while self.listIndex < len(self.listOfList):
            if self.elemIndex < len(self.listOfList[self.listIndex]:
                return True
            else:
                self.listIndex += 1
                self.elemIndex = 0
            return False

    
if __name__ == "__main__":
    solution = Solution()

File: src/CSharp/251.Flatten2DVector.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;

namespace PeefyLeetCode.Flatten2DVector
{

    public class Vector2D
    {
        List<List<int>> listOfList;
        int listIndex;
        int elemIndex;
        public Vector2D(List<List<int>> vec2d)
        {
            listOfList = vec2d;
            listIndex = 0;
            elemIndex = 0;
        }

        public int Next()
        {
            return listOfList[listIndex][elemIndex++];
        }

        public bool HasNext()
        {
            while (listIndex < listOfList.Count)
            {
                if (elemIndex < listOfList[listIndex].Count)
                {
                    return true;
                }
                else
                {
                    listIndex++;
                    elemIndex = 0;
                }
            }
            return false;
        }
    }

}